tests/mail/tests.py

```diff
@@ -894,6 +894,21 @@ def test_dont_mangle_from_in_body(self):
         )
         self.assertNotIn(b">From the future", email.message().as_bytes())
 
+    def test_long_address_folding(self):
+        # Ticket #35378
+        # should encode long addresses with special characters using
+        # 7bit Content-Transfer-Encoding
+        msg = EmailMessage(
+            "Subject",
+            "Long address with special characters",
+            "from@example.com",
+            ['"Người nhận a very very long, name" <to@example.com>'],
+        )
+        s = msg.message().as_bytes()
+        self.assertIn(b"Content-Transfer-Encoding: 7bit", s)
+        s = msg.message().as_string()
+        self.assertIn("Content-Transfer-Encoding: 7bit", s)
```

Comment:
I reverted your changes from `message.py` and was expecting the this test to fail but surprisingly it passed.
The main issues mentioned in the ticket was the incorrect folding of long address headers when using 7bit Content-Transfer-Encoding , So if you'll check for the `Content-Transfer-Encoding: 7bit` in message string it will always be there. 

After you set `maxheaderlen=0`, it will treat all the long addresses as one string and there wont be any new line (\n) in between the address. So, in my opinion it will be better if you check the encoded long address for a new line symbol in somewhere between the address. If its not there then it means that its no longer folding the long addresses and treating it as one single string. Does that make sense to you?
```suggestion
        s = msg.message().as_bytes()
        self.assertIn(b"=?utf-8?b?TmfGsOG7nWkgbmjhuq1uIGEgdmVyeSB2ZXJ5IGxvbmcsIG5hbWU=?= <to@example.com>\n", s)
        s = msg.message().as_string()
        self.assertIn("=?utf-8?b?TmfGsOG7nWkgbmjhuq1uIGEgdmVyeSB2ZXJ5IGxvbmcsIG5hbWU=?= <to@example.com>\n", s)
```

Reply:
Yes, a test update is required 👍
Something like this might work
```python
    def test_long_address_folding(self):
        msg = EmailMessage(
            "Subject",
            "Long address with special characters",
            "from@example.com",
            ['"Người nhận a very very long, name" <to@example.com>'],
        ).message()
        msg.policy = policy.default.clone(cte_type="7bit")
        self.assertIn(
            "=?utf-8?b?TmfGsOG7nWkgbmjhuq1uIGEgdmVyeSB2ZXJ5IGxvbmcs?= name <to@example.com>",
            msg.as_bytes().decode("ascii")
        )
```
This would use the [example from the discussion](https://github.com/anymail/django-anymail/issues/369#issuecomment-2053731851).

This also fails and I am not sure adding `maxheaderlen=0` is the right solution. It goes against previous design decisions (see ticket-31784) and [RFC 2822](https://datatracker.ietf.org/doc/html/rfc2822#section-2.1.1) stating that 
> Each line of characters MUST be no more than 998 characters, and SHOULD be no more than 78 characters, excluding the CRLF.

Allowing longer line length was discussed and discouraged in [this comment](https://github.com/anymail/django-anymail/issues/369#issuecomment-2052683478).
> Allowing longer lines would solve the problem (no folding needed), but has similar compatibility concerns to using 8bit (or worse).

In short, I think more investigation is needed 🤔